payu/calendar.py

```diff
@@ -90,6 +90,68 @@ def calculate_leapdays(init_date, final_date):
     return datetime.timedelta(days=leap_days)
 
 
+# TODO: The caltype logic could be simplified if we switched
+# to using just a string as the caltype input. Might require reworking of other
+# functions for consistency.
+def seconds_between_dates(start_date, end_date, caltype_int):
+    """
+    Calculate the number of seconds between two datetime objects
+    with a specified calender type by using cftime datetime objects
+    as intermiaries.
+
+    Parameters
+    ----------
+    start_date: datetime.date
+    end_date: datetime.date
+    caltype: Integer, either GREGORIAN or NOLEAP
+
+    Returns
+    -------
+    seconds: Number of seconds between start_date and end_date.
+    """
+    # Get the cftime string corresponding to the caltype integer
+
+    # TODO: Is it confusing that GREGORIAN means proleptic gregorian?
+    if caltype_int == GREGORIAN:
+        calendar_str = "proleptic_gregorian"
+    elif caltype_int == NOLEAP:
+        calendar_str = "noleap"
```

Comment:
If this is just a case change, then I'd just do a blanket `.lowercase()` at the beginning to ensure any calendar is correct.

Reply:
This one's actually converting from the [integer constants]( https://github.com/ACCESS-NRI/payu/blob/8f18f245787cafa940a9c397b5173479a40b6273/payu/calendar.py#L7) used in the cice namelists to the calendar strings used by `cftime`.

Reply:
Ah duh! (Me). Sorry.
```suggestion
    elif caltype_int == NOLEAP:
        calendar_str = "noleap"
```
```suggestion
    elif caltype_int == NOLEAP:
        calendar_str = "noleap"
```
```suggestion
    elif caltype_int == NOLEAP:
        calendar_str = "noleap"
```
```suggestion
    elif caltype_int == NOLEAP:
        calendar_str = "noleap"
```

Reply:
Ah duh! (me). Sorry.